sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -367,13 +442,177 @@ namespace Azure { namespace Storage { namespace DataLake {
   };
 
   /**
-   * @brief Optional parameters for DataLakePathClient::Create
+   * @brief Optional parameters for DirectoryClient::Delete
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/delete
+   */
+  struct PathDeleteOptions
+  {
+    /**
+     * @brief Context for cancelling long running operations.
+     */
+    Azure::Core::Context Context;
+
+    /**
+     * @brief When deleting a directory, the number of paths that are deleted with each invocation
+     *        is limited. If the number of paths to be deleted exceeds this limit, a continuation
+     *        token is returned in this response header.  When a continuation token is returned in
+     *        the response, it must be specified in a subsequent invocation of the delete operation
+     *        to continue deleting the directory.
+     */
+    Azure::Core::Nullable<std::string> Continuation;
+
+    /**
+     * @brief Required and valid only when the resource is a directory. If "true", all paths beneath
+     *        the directory will be deleted. If "false" and the directory is non-empty, an error
+     *        occurs.
+     */
+    Azure::Core::Nullable<bool> RecursiveOptional;
```

Comment:
```suggestion
    Azure::Core::Nullable<bool> Recursive;
```

Reply:
We need to be consistent, right?

Reply:
I don't think it is the same thing. In `DirectoryDeleteOptions` it is not optional.

Reply:
So why do we need `Optional` suffix here? What's the difference between `Recursive` and `RecursiveOptional`?

Reply:
After offline discussion, decided to make recursive mandatory in `DirectoryClient` and removed `OptionalSuffix` here.